Classes/RedirectStorage.php

```diff
@@ -138,17 +149,19 @@ protected function addRedirectByHost($sourceUriPath, $targetUriPath, $statusCode
     {
         $redirect = new Redirect($sourceUriPath, $targetUriPath, $statusCode, $host);
         $this->updateDependingRedirects($redirect);
+        $this->persistenceManager->persistAll();
```

Comment:
Yes, still not optimal but at least it would only persist whitelisted objects then